components/layout_2020/flexbox/layout.rs

```diff
@@ -301,19 +309,65 @@ impl FlexContainer {
         let line_interval = match flex_context.container_definite_inner_size.cross {
             Some(cross_size) if line_count >= 2 => {
                 let free_space = cross_size - content_cross_size;
+                let layout_is_flex_reversed = flex_context.flex_wrap_reverse;
 
                 cross_start_position_cursor = match flex_context.align_content {
+                    AlignContent::Start => Length::zero(),
+                    AlignContent::FlexStart => {
+                        if layout_is_flex_reversed {
+                            free_space
+                        } else {
+                            Length::zero()
+                        }
+                    },
+                    AlignContent::End => free_space,
+                    AlignContent::FlexEnd => {
+                        if layout_is_flex_reversed {
+                            Length::zero()
+                        } else {
+                            free_space
+                        }
+                    },
                     AlignContent::Center => free_space / 2.0,
-                    AlignContent::SpaceAround => free_space / (line_count * 2) as CSSFloat,
-                    AlignContent::FlexEnd => free_space,
-                    _ => Length::zero(),
+                    AlignContent::Stretch => Length::zero(),
+                    AlignContent::SpaceBetween => Length::zero(),
+                    AlignContent::SpaceAround => {
+                        if free_space >= Length::zero() {
+                            (free_space / line_count as CSSFloat) / 2.0
+                        } else {
+                            free_space / 2.0
+                        }
+                    },
+                    AlignContent::SpaceEvenly => {
+                        if free_space >= Length::zero() {
+                            free_space / (line_count + 1) as CSSFloat
+                        } else {
+                            free_space / 2.0
+                        }
+                    },
                 };
 
-                match flex_context.align_content {
-                    AlignContent::SpaceBetween => free_space / (line_count - 1) as CSSFloat,
-                    AlignContent::SpaceAround => free_space / line_count as CSSFloat,
-                    _ => Length::zero(),
-                }
+                // Note: free_space may be negative in which case we wish the gaps between items to be zero
+                // (although this will change once gap is implemented)
+                // TODO: Implement gap property
+                let item_main_interval = if free_space <= Length::zero() || line_count == 0 {
```

Comment:
Yeah, I think we'd better keep the condition here as the line_count >= 2 seems liable to be removed. Happy to change it to `<= 1` if you think that's clearer. Devide by zero is definitely wrong here as we'd end up with `NaN` rather than `0`.